include/aspect/simulator.h

```diff
@@ -1428,21 +1428,29 @@ namespace aspect
 
 
       /**
-       * Add constraints to the given @p constraints object that are required
-       * for unique solvability of the velocity block based on the nullspace
-       * removal settings.
+       * Add constraints to the given @p constraints object that are
+       * required for unique solvability of the velocity block based
+       * on the nullspace removal settings.
        *
-       * This method will add a zero Dirichlet constraint for the first
-       * velocity unknown in the domain for each velocity component, which is
-       * later being processed for translational or linear momentum removal.
-       * This avoids breakdowns of the linear solvers that otherwise occurred
-       * in some instances.
+       * For translation, this method will add a zero Dirichlet
+       * constraint for the first velocity unknown in the domain for
+       * each velocity component, which is later being processed for
+       * translational or linear momentum removal.  This avoids
+       * breakdowns of the linear solvers that otherwise occurred in
+       * some instances.
        *
-       * @note: Rotational modes are currently not handled and don't appear to
-       * require constraints so far.
+       * For rotation, we fix velocity components in 1 or 2 points on
+       * the surface of the sphere/spherical shell in 2d and 3d,
+       * respectively.
        */
       void setup_nullspace_constraints(AffineConstraints<double> &constraints);
 
+      /**
+       * Like above, but on a given multigrid level @p level.
+       */
+      void setup_nullspace_constraints(AffineConstraints<double> &constraints,
+                                       const DoFHandler<dim> &dof_handler,
+                                       const int level);
```

Comment:
Out of curiosity, why do you need to pass a `DoFHandler` argument to the second function but not the first?